### ISSUE TITLE
Add Order Number to automatic notification email that goes to supplier when order cycles closes

### ISSUE THREAD

**Issue by karlafalk** [openfoodfoundation/wishlist#524]
## 
Hub would like to have suppliers to be able to see Order Numbers (along with Customer Names) in automatic notification email that goes to a supplier when the order cycle closes. 

This would help with identify, tracking and reconciling of payments between supplier and hub, which can sometimes happen weeks after the order has taken place, and can be confusing if the same customer has orders over multiple weeks.  Using Order Numbers reduces confusion around orders that have been fulfilled and are being invoiced to the hub.    

## 
- Hub Managers (MCFEs)
- Producers

## Potential solutions that could solve the problem?
Right now hub managers are having to send Order Numbers to suppliers outside of OFN from an OC Customers Totals report.  If this info was included in the automatic notification email then it would cut down on the admin time for hub managers.  

<img width="1209" alt="Screenshot 2024-10-28 at 9 06 21 AM" src="https://github.com/user-attachments/assets/3ab61b9e-1df0-4293-9396-ed04b6ffd532">

**Comment by RachL** [comment]
@tschumilas  here

> to be able to see Order Numbers (along with Customer Names)

I think this means that we would link order numbers to the "display customer names" setting in shop preferences. Is that a correct assumption?

Otherwise the email does not group by customers. Also note that the email would become longer as we would need to create 1 line per order, instead of grouping by customers like we currently do. I wonder if this can cause confusion sometimes when packing :thinking: 

Maybe we need to sort the table by customer.

**Comment by tschumilas** [comment]
I think this issue (@karlafalk  please chime in)  is to put the order number on the email sent to suppliers next to the customer's name,  similar to how we are putting the customer code.   But,  are you thinking they won't both (code and order number) won't fit?
And,  I guess we have the issue of what if a customer placed more than 1 order to the same supplier for the same product (my comment about the pine cones in the example below)

Options?

![image](https://github.com/user-attachments/assets/3fcaf595-4606-4294-a60a-bc6af06c2eb3)

**Comment by karlafalk** [comment]
Yes @RachL, the ability for producers to view order numbers would be linked to the 'display customer names' setting in shop preferences.  

Yes that is correct @tschumilas.  Order number would be placed beside the customer name on the email notification sent to producers when the order cycle closes.  It shouldn't lengthen the email all that much as there is already one line per customer/product, which would correspond to an order.  It should only be longer if a customer has placed multiple orders with the same product, as Theresa mentioned above, and the question would be in that case does more than one order number get placed beside the name or does an additional row get created for additional order #s (with the same product and customer name as a previous row)?   

@tschumilas were you also looking for customer contact info to be included some way in the automatic notification email?

**Comment by tschumilas** [comment]
I was looking for customer code -  which is what flower hubs need.  Because these are wholesale BtoB  hubs,  they use 'code' for the business name of the buyer.  The suppliers pack based on that.  So this would save them running a report.
Buyer contact was discussed when we first introduced this notification as i recall.  I think we felt that room on the email was an issue.  Maybe there was also security concerns with contact info -  phone numbers in particular -  being sent in emails?

One option maybe -  if per line space is an issue -  we could use just one name -  likely last name.    @RachL - I'm just going to do a little post in instance managers just to see what others are thinking about these nofications now that we seem into it....

**Comment by RachL** [comment]
For the record, the discussion on slack is here: https://openfoodnetwork.slack.com/archives/CG7NJ966B/p1734011373858429

@tschumilas I'm thinking we should tidy up a bit and I was thinking of the following:

- Display to everyone customer code and order number: would that be a problem? I feel these info are rather anonymous, but maybe I'm missing something
- Keep current setting to impact displaying names or not in report & email
- Add a setting to reveal customer credentials in report & email

These 2 lasts point would help EU instances to avoid sharing credentials, while still sharing names.

Last topic would be display of these info, especially on mobile, but we can leave this for a later stage, let's first agree on the content.

**Comment by tschumilas** [comment]
This sounds good for me (for flower hubs at least) @karlafalk  - I think this would address the request you made also?

Thanks for this @RachL

**Comment by karlafalk** [comment]
This sounds good to me @RachL and @tschumilas.  I think this would address both inquires I had recently from hubs about this report, and allow the most flexibility.  

I am assuming when you say customer credentials @RachL that means customer contact info?  Would this be phone, email or both?   

Thanks!!

**Comment by RachL** [comment]
Sorry for being late here. Yes @karlafalk I was thinking about email and phone numbers.

issues created:

https://github.com/openfoodfoundation/openfoodnetwork/issues/13128
https://github.com/openfoodfoundation/openfoodnetwork/issues/13129
https://github.com/openfoodfoundation/openfoodnetwork/issues/13130

closing here :)
